hardhat/Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM node:16.9.1 as dependency-base
+
+ENV APP_ROOT /app
+
+RUN mkdir ${APP_ROOT}
+WORKDIR ${APP_ROOT}
+ADD . ${APP_ROOT}
+
+RUN npm ci
```

Comment:
Please:
- first COPY `package.json` and `package-lock.json`
- then install dependencies (`RUN npm ci`)
- only then COPY the rest `COPY . ${APP_ROOT}`

This will let docker keep cache of the installed packages and not rerun `npm ci` in case `package-lock.json` hasn't changed

Also, [prefer `COPY` above `ADD`](https://stackoverflow.com/a/24958548)